src/datumaro/cli/util/compare.py

```diff
@@ -83,22 +82,24 @@ def save(self, a: IDataset, b: IDataset):
         if len(a) != len(b):
             print("Datasets have different lengths: %s vs %s" % (len(a), len(b)))
 
-        a_classes = a.categories().get(AnnotationType.label, LabelCategories())
-        b_classes = b.categories().get(AnnotationType.label, LabelCategories())
-        class_mismatch = [
-            (idx, a_cls, b_cls)
-            for idx, (a_cls, b_cls) in enumerate(zip_longest(a_classes, b_classes))
-            if getattr(a_cls, "name", None) != getattr(b_cls, "name", None)
+        a_classes = set(
+            a_cls.name for a_cls in a.categories().get(AnnotationType.label, LabelCategories())
+        )
+        b_classes = set(
+            b_cls.name for b_cls in b.categories().get(AnnotationType.label, LabelCategories())
+        )
```

Comment:
Would you please make it cleaner by introducing some interface in the `Category` class such as
```suggestion
        a_classes = set(a.categories().get_label_cat_names())
        b_classes = set(b.categories().get_label_cat_names())
```

Reply:
F.Y.I., if you add the interface, it should be in the [`CategoriesInfo`](https://github.com/openvinotoolkit/datumaro/blob/44c1391abd7ef554b89615b422717833634174d7/src/datumaro/components/dataset_base.py#L58C1-L58C50), not in `Category`.

Reply:
Fix this through define `get_label_cat_names` function in `Dataset`.